### ISSUE TITLE
making the package dynamic

### ISSUE THREAD

**Issue by darvishiyan-mahdi** [Zizaco/entrust#936]
first of all i want to thank you because of your awsome package
i want to know is there anyway to make it dynamic
for example i deploy the system and the user of system want to define a new role in system i want to know is there any way that the system handle the role and route by itself with out coding?

**Comment by shkrwnd** [comment]
Well I think the current  package is sufficient if the user wants to create role or permission dynamically without coding (I think you meant front end) 
 Just give the required details to create role or permissions (in front end)  and from backend use the same Role::create() and Permission::create() methods or you can directly insert the data into necessary tables(permissions, roles , permission_role and role_user) if you know how they work.They are just simple relationships like  joins.

**Comment by darvishiyan-mahdi** [comment]
> Well I think the current package is sufficient if the user wants to create role or permission dynamically without coding (I think you meant front end)
> Just give the required details to create role or permissions (in front end) and from backend use the same Role::create() and Permission::create() methods or you can directly insert the data into necessary tables(permissions, roles , permission_role and role_user) if you know how they work.They are just simple relationships like joins.


i know the mechanism but i like to know how does it handle in the role

**Comment by shkrwnd** [comment]
> > Well I think the current package is sufficient if the user wants to create role or permission dynamically without coding (I think you meant front end)
> > Just give the required details to create role or permissions (in front end) and from backend use the same Role::create() and Permission::create() methods or you can directly insert the data into necessary tables(permissions, roles , permission_role and role_user) if you know how they work.They are just simple relationships like joins.
> 
> i know the mechanism but i like to know how does it handle in the role

So you want to know how it works internally.

**Comment by darvishiyan-mahdi** [comment]
> > > Well I think the current package is sufficient if the user wants to create role or permission dynamically without coding (I think you meant front end)
> > > Just give the required details to create role or permissions (in front end) and from backend use the same Role::create() and Permission::create() methods or you can directly insert the data into necessary tables(permissions, roles , permission_role and role_user) if you know how they work.They are just simple relationships like joins.
> > 
> > 
> > i know the mechanism but i like to know how does it handle in the role
> 
> So you want to know how it works internally.

i want to know when  i deploy the system and they make a new role in it does entrust can handle it by itself or not?

**Comment by shkrwnd** [comment]
> > > > Well I think the current package is sufficient if the user wants to create role or permission dynamically without coding (I think you meant front end)
> > > > Just give the required details to create role or permissions (in front end) and from backend use the same Role::create() and Permission::create() methods or you can directly insert the data into necessary tables(permissions, roles , permission_role and role_user) if you know how they work.They are just simple relationships like joins.
> > > 
> > > 
> > > i know the mechanism but i like to know how does it handle in the role
> > 
> > 
> > So you want to know how it works internally.
> 
> i want to know when i deploy the system and they make a new role in it does entrust can handle it by itself or not?

Yes it does handle it by itself .You can make any role or permission after deploying your application as it directly queries the database  and stores it in cache.

**Comment by darvishiyan-mahdi** [comment]
Yes it handle the role but does it handle the route by it self?
